### PR TITLE
fix: clear stale task state after task removal

### PR DESCRIPTION
**Pull request by rev2ret**
## Summary
- clear task-scoped UI state when deleting the currently open task
- clear current task and related derived state when task history is wiped
- add integration coverage for both the deleted-current-task case and the unrelated-task case

## Testing
- `corepack pnpm --filter @accomplish/web exec vitest run __tests__/integration/renderer/taskStore.integration.test.ts --config vitest.integration.config.ts`
- `corepack pnpm --filter @accomplish/web typecheck`

## Notes
- `corepack pnpm --filter @accomplish/web test:integration -- taskStore.integration.test.ts` still booted unrelated suites and hit existing Windows asset-path failures outside this change.

## Summary by CodeRabbit

* **Bug Fixes**
  * Deleting a task reliably clears all task-scoped metadata and leaves other tasks intact.
  * Clearing history fully resets current task, task lists, permission requests, and scoped defaults.
  * Progress or completions from stale/untracked tasks are ignored to prevent incorrect state updates.

* **Tests**
  * Added integration tests covering ignored late completions, scoped-state clearing on delete/clear, and preservation of active task state when deleting other tasks.

### CODE REVIEW

**Comment by orcaman** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/or.hiltch/pr-checks/28d65cd5-9993-4935-8ed8-74ae778dc85f?refs=e8e20c94dde7b123533aa323779799405bc03712&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/or.hiltch/pr-checks/28d65cd5-9993-4935-8ed8-74ae778dc85f/license?refs=e8e20c94dde7b123533aa323779799405bc03712&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

**Review by mavishay — APPROVED** [review]
🟢 LGTM

File: apps/web/src/client/stores/taskStore.ts [hunk 1]
```diff
@@ -103,6 +103,77 @@ interface TaskState {
...
+    permissionRequests: {},
+    setupProgress: null,
+    setupProgressTaskId: null,
+    setupDownloadStep: 1,
+    startupStage: null,
+    startupStageTaskId: null,
+    todos: [],
+    todosTaskId: null,
+  };
+}
```
[the remark this answers is not in the document]
Comment (mavishay) [reply apps/web/src/client/stores/taskStore.ts:194]: @rev2ret Can you address this please 🙏🏽

**Review by mavishay — APPROVED** [review]
🟢 LGTM Ready to merge

**Comment by mavishay** [comment]
## 🙏 Thank you, @rev2ret!

Great bug fix! Your changes to `taskStore.ts` cleanly handle the edge cases of stale task state — clearing task-scoped UI when deleting the current task, resetting state on history wipe, and ignoring late async updates from removed tasks.

The integration tests you added (+276 lines) give solid coverage for all three scenarios. This is now live in main. 🎉
